### ISSUE TITLE
How to enable multi thread computing for tf.data

### ISSUE THREAD

**Issue by yc0619** [tensorflow/tensorflow#52381]
System: Ubuntu 20.04
GPU/CUDA: RTX3090 CUDA10.1(must be upgraded)
Before Dataloader the data doesn't goes to the GPU. And i have read about, that df.data runs on cpu. And i have tried:

```
tf.config.threading.set_intra_op_parallelism_threads(64)
tf.config.threading.set_inter_op_parallelism_threads(64)
```
before the code:
`train_dataset = tf.data.Dataset.from_tensor_slices((train_data, train_lbl))`
Since i ran the line, it always come to such warning:
`W tensorflow/core/framework/cpu_allocator_impl.cc:80] Allocation of 7957118976 exceeds 10% of free system memory.`
Is that because always max. 2 threads are used. (I have 64 threads cpu). Thanks for the help!

**Comment by tilakrayal** [comment]
@yc0619 ,
In order to expedite the trouble-shooting process, could you please provide a complete code  and the TensorFlow version you are using.Thanks!

**Comment by yc0619** [comment]
Sorry i am not allowed to give you the entire code. But i can tell you the workflow (these codes are right)
1. in csv: data label, image path
2. read csv to get dataframe, append image in dataframe
3. from datafram get data, lbl and use tf.data.Dataset.from_tensor_slices((data, lbl))
And in the third step, just 2 of cpu threads are used, but when i give the data and lbl (np.array) in model.fit, it could be run, but at validation, cpu are heavily used (2 of the 64 threads are used). So i would like to ask if there is some methode to enable me use other 10 or 20 more threads for the code.

best regards! :)

**Comment by tilakrayal** [comment]
Also please take a look at this SO [link](https://stackoverflow.com/questions/47086599/parallelising-tf-data-dataset-from-generator) with the similar information.It helps.If you are unable to provide the reproducible code, please post this issue in tf discussion as this is not either feature or bug request.Thanks
